test/http_download_test.rb

```diff
@@ -42,6 +44,12 @@ def test_regular_call_with_ssl_verify
     assert_equal expected, Proxy::HttpDownload.new('src', 'dst', verify_server_cert: true).command
   end
 
+  def test_retrieving_404
+    thread = Proxy::HttpDownload.new('http://example.com/404', '/tmp/404', verify_server_cert: true).start
```

Comment:
This reaches out to the internet, which is a bad idea. It would be better to start some HTTP server locally and use that.